backend/src/main/java/com/beaplab/BeaplabEngine/controller/WatchController.java

```diff
@@ -0,0 +1,336 @@
+                    "A sample ajax request is as follows:\n\n" +
+                    "--------begin example-------\n" +
+                    "var formData = new FormData();\n" +
+                    // not sure how to do this with general controller...
+                    "formData.append('fname', 'applewatch.zip');\n" +
+                    "formData.append('data', blob);\n" +
+                    "$.ajax({\n" +
+                    "type: 'POST',\n" +
+                                                // not sure here either
+                    "                            url: '/rest/beapengine/applewatch/upload',\n" +
+                    "                            data: formData,\n" +
+                    "                            processData: false,\n" +
+                    "                            contentType: false,\n" +
+                    "                            success: function (data, status, xhr) {\n" +
+                    "                                console.log('status: ' + status + ', data: ' + data);\n" +
+                    "                            },\n" +
+                    "                            error: function (jqXhr, textStatus, errorMessage) {\n" +
+                    "                                var result = JSON.parse(jqXhr.responseText);\n" +
+                    "                                console.log(\"error: \" + result)\n" +
+                    "                            }\n" +
+                    "                        });\n" +
+                    "--------end example-------\n",
+                    type = "POST",
+                    required = true, paramType = "blob")
+    })
+    public ResponseEntity<JSONObject> upload(
+        @ApiParam(value = "The watch type to be uploaded", required = true)
+        @PathVariable("watchType") String watchType,
+        @ApiIgnore MultipartHttpServletRequest request) {
+
+        rollbar.info("in WatchController/upload");
+
+        HttpSession session = request.getSession(false);
+        if(session == null || !request.isRequestedSessionIdValid()) { // invalid session
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Invalid session");
+            jsonObject.put("status_code", HttpStatus.NON_AUTHORITATIVE_INFORMATION.value());
+
+            return new ResponseEntity<>(jsonObject, (HttpStatus.valueOf((int)jsonObject.get("status_code"))));
+        }
+
+
+        if(watchType != "applewatch" && watchType != "fitbit"){
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Invalid watch type in url");
+            // wrong status_code
+            jsonObject.put("status_code", 404);
+            return new ResponseEntity<>(jsonObject, (HttpStatus.valueOf((int)jsonObject.get("status_code"))));
+        }
```

Comment:
Any recommendations for adding status codes? They should probably be the same across the whole controller right?

For example, an invalid watch type should always have status code 501

Reply:
This is a great resource for following conventions on status codes. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

In this case though, a non matching watchType means a mistake from the client in using the route. In this case a status code `400` which is a `Bad request` will be more appropriate.